File: Travel/Travel.Models/RouteTicket/RouteProfitReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Travel.Models.Enums;

namespace Travel.Models.RouteTicket
{
    public class RouteProfitReport
    {
        public long RouteId { get; set; }
        public string RouteName { get; set; }
        public BusType BusType { get; set; }
        public int TicketsSold { get; set; }
        public double TotalRevenue { get; set; }
        public double TotalCost { get; set; }
        public double TotalProfit { get; set; }
    }

}

File: Travel/UI/travel_admin/lib/model/route_ticket/route_profit_report.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';

part 'route_profit_report.g.dart';

@JsonSerializable()
class RouteProfitReportModel {
  int? routeId;
  String? routeName;
  double? totalProfit;
  int? busType;
  int? ticketsSold;
  double? totalRevenue;
  double? totalCost;
  RouteProfitReportModel(this.routeId, this.routeName, this.totalProfit,
      this.busType, this.ticketsSold, this.totalCost, this.totalRevenue);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory RouteProfitReportModel.fromJson(Map<String, dynamic> json) =>
      _$RouteProfitReportModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$RouteProfitReportModelToJson(this);
}
